main.c

```diff
@@ -1077,6 +1077,10 @@ int main(int argc, char **argv)
 	}
 
 	tcmu_dbg("handler path: %s\n", handler_path);
+
+	tcmu_block_netlink();
+	tcmu_reset_netlink();
```

Comment:
May be tcmu_reset_netlink() should be executed only if tcmu_block_netlink() is successful ?

Reply:
It will fail too if the block_netlink is failed, so here IMO is not a must.

Reply:
I was thinking of a case where tcmu_block_netlink() failed in tcmu_set_cfgfs_ul->tcmu_set_cfgfs_str(), but I will leave that up to you :-)

Reply:
Yeah, for both I am okay, currently this just as the reset ring does by @mikechristie.

Reply:
It's probably fine. In the unsupported case you get an extra warning about both block and reset. For the case where the write  fails for the block then best case we avoid races and it works ok for the reset and worst case someone is doing a rtstlib call at the same time and we race and you are in the same hang as before.